Fix #357: Scrollbar margin inconsistency

SammysHP opened the pull request:
Completely refactored scrollbar and added proper stepper support (although they are disabled by default).

Examples with and without steppers:

![scrollbar_steppers](https://cloud.githubusercontent.com/assets/922874/14609828/be8b86a2-058b-11e6-8138-aed514ae65c6.gif) ![scrollbar](https://cloud.githubusercontent.com/assets/922874/14609837/c399442c-058b-11e6-862e-9437b01ed9c9.gif)

SammysHP committed:
Fix #357: Scrollbar margin inconsistency

Completely refactored scrollbar and added proper stepper support
(although they are disabled by default).

khurshid-alam commented on gtk-3.20/scss/widgets/_scrollbar.scss:
```diff
@@ -27,135 +29,108 @@
-    treeview ~ scrollbar.vertical {
-        border-top: 1px solid $borders_color;
-        margin-top: -1px;
-    }
```
Why is this removed?

SammysHP replied:
It added a border where no border is necessary (because there is already a border from the treeview).

khurshid-alam commented:
Steppers are obsolete for overlay scrollbar. We should go without it.

SammysHP commented:
By default they are disabled. But they can be enabled via a custom gtk.css in .config/gtk3.0 and without this code there would be only gray boxes and an ugly scrollbar. Thus I thought it would be wise to keep some code as fallback.

SammysHP commented:
Shall I fix the style issues? (Although I think that single line declarations are absolutely fine for a single element.)

megax commented:
@SammysHP Minimal design correction :) :
$_slider_min_length: 40px; (increased usability)
Now (20 px):
![image](https://cloud.githubusercontent.com/assets/461493/14760578/1dc80554-0948-11e6-97ec-b4ab3bf2197d.png)

New:
![image](https://cloud.githubusercontent.com/assets/461493/14760580/22bb4490-0948-11e6-9a25-461d5967499f.png)

$_slider_small_width: 4px; (Old Numix (circa size))

SammysHP commented:
@megax 
Will do it in ~5h.

SammysHP commented:
Applied changes suggested by @megax, fixed some of the doggy issues (but still kept single declarations in one line because I'm of the opinion that this way is much more readable) and rebased on current master.

khurshid-alam commented:
@SammysHP @megax Did any of you create a pull for fixing terminal (vte) scrollbar before?

SammysHP commented:
@khurshid-alam See https://github.com/numixproject/numix-gtk-theme/issues/359#issuecomment-213799629.

megax commented on gtk-3.20/scss/widgets/_scrollbar.scss:
```diff
@@ -27,135 +29,106 @@
+                    min-height: $_slider_small_width;
                     background-color: $fg_color;
-                    border: 1px solid if($variant == 'light', white, black);
+                    border: 1px solid if($variant == "light", white, black);
```
No convert `$variant == 'light'` to `$variant == "light"`
white and black convert $balck, $white :)

SammysHP replied:
Why single quotes instead of double quotes? The dog asked me to change single quotes into double quotes. :(

I'll fix the colors tomorrow.

megax replied:
No bot comment =/ https://github.com/numixproject/numix-gtk-theme/pull/394/files#diff-5b92b8ba7bbabb66d6af49b3d5e3f220R70

SammysHP replied:
As you already fixed it in another pull request, this will/can be done during merge.
